pkg/lifecycle/image_updater.go

```diff
@@ -0,0 +1,106 @@
+package lifecycle
+
+import (
+	"io"
+	"path"
+
+	ggcrv1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/pivotal/kpack/pkg/registry/imagehelpers"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	k8s "k8s.io/client-go/kubernetes"
+
+	"github.com/pivotal/build-service-cli/pkg/commands"
+	buildk8s "github.com/pivotal/build-service-cli/pkg/k8s"
+	"github.com/pivotal/build-service-cli/pkg/registry"
+)
+
+const (
+	lifecycleImageName     = "lifecycle"
+	lifecycleMetadataLabel = "io.buildpacks.lifecycle.metadata"
+)
+
+type preUpdateHook func(*corev1.ConfigMap)
+
+type ImageUpdater interface {
+	UpdateImage(srcImgStr string, tlsCfg registry.TLSConfig, hooks ...preUpdateHook) (*corev1.ConfigMap, error)
+}
+
+type imageUpdater struct {
+	dryRun      bool
+	writer      io.Writer
+	fetcher     registry.Fetcher
+	relocator   registry.Relocator
+	kpCfgHelper buildk8s.ConfigHelper
+	k8sClient   k8s.Interface
+}
+
+func NewImageUpdater(
```

Comment:
Similar to the changeSummarizer is this extra object needed?

Reply:
addressed in https://github.com/vmware-tanzu/kpack-cli/commit/e09d755bd1759ef19b29f89b61834db58c048b22